src/app/genre/genre.component.ts

```diff
@@ -1,23 +1,44 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit,OnDestroy } from '@angular/core';
+import { NavigationStart, RouteConfigLoadEnd, Router } from '@angular/router';
+import { filter, Subject, takeUntil } from 'rxjs';
 import { SampleServiceService } from '../sample-service.service';
 
 @Component({
   selector: 'app-genre',
   templateUrl: './genre.component.html',
   styleUrls: ['./genre.component.css']
 })
-export class GenreComponent  implements OnInit{
+export class GenreComponent  implements OnInit,OnDestroy{
+  OnDestroy$=new Subject<boolean>
 
   genreData:any
-  constructor(private service:SampleServiceService){}
+  Comedy:any
+  Thriller:any
+  Romance:any
+
+  constructor(private service:SampleServiceService , private route:Router){}
+  ngOnInit():void{  
+    this.route.events.pipe(filter(navigate=>navigate instanceof NavigationStart)).pipe(takeUntil(this.OnDestroy$)).subscribe((urlLink:any)=>{
+      console.log('----',urlLink);
```

Comment:
why is the console not yet removed and you have resolved the comment without making changes...?